sdk/monitor/monitor-opentelemetry-exporter/src/platform/nodejs/httpSender.ts

```diff
@@ -34,9 +37,29 @@ export class HttpSender implements Sender {
    * @internal
    */
   async send(envelopes: Envelope[]): Promise<SenderResult> {
+    let options: ApplicationInsightsClientTrackOptionalParams = {};
     try {
-      const { _response: res } = await this._appInsightsClient.track(envelopes);
-      return { statusCode: res.status, result: res.bodyAsText ?? "" };
+      let response: FullOperationResponse | undefined;
+      function onResponse(rawResponse: FullOperationResponse, flatResponse: unknown): void {
+        response = rawResponse;
+        if (options.onResponse) {
+          options.onResponse(rawResponse, flatResponse);
+        }
+      }
+      await this._appInsightsClient.track(envelopes, {
+        ...options,
+        onResponse
+      });
+
+      if (response?.status === 404) {
+        throw new RestError(response?.bodyAsText!, {
+          statusCode: response?.status,
+          request: response?.request,
+          response: response
+        });
+      }
```

Comment:
Do you really have to throw yourself here? I think `core-client` would throw before the call at line 49 finishes but I could be wrong.

Reply:
You are right, 404 is handled by core client. Add a test case and remove this in next udpate.